pandas/io/formats/style.py

```diff
@@ -1167,6 +1163,9 @@ def _apply(
                         f"Expected shape: {data.shape}"
                     )
                 result = DataFrame(result, index=data.index, columns=data.columns)
+        else:
+            axis = self.data._get_axis_number(axis)
+            result = data.T.apply(func, axis=0, **kwargs).T if axis == 1 else data.apply(func, axis=0, **kwargs)
```

Comment:
would prefer to keep same pattern as above in case DataFrame.apply gets fixed so the transpose can be refactored.

```suggestion
            if axis == 0:
                result = data.apply(func, axis=0, **kwargs)
            else:
                result = data.T.apply(func, axis=0, **kwargs).T  # see GH 42005
```